src/core/src/pass/serialize.cpp

```diff
@@ -1229,21 +1229,21 @@ bool pass::Serialize::run_on_model(const std::shared_ptr<ov::Model>& model) {
         if (xmlDir != m_xmlPath)
             ov::util::create_directory_recursive(xmlDir);
 
-        std::ofstream bin_file(m_binPath, std::ios::out | std::ios::binary);
-        OPENVINO_ASSERT(bin_file, "Can't open bin file: \"" + m_binPath + "\"");
-
-        // create xml file
-        std::ofstream xml_file(m_xmlPath, std::ios::out);
-        OPENVINO_ASSERT(xml_file, "Can't open xml file: \"" + m_xmlPath + "\"");
-
         try {
+            // create bin file
+            std::ofstream bin_file(m_binPath, std::ios::out | std::ios::binary);
+            bin_file.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+            OPENVINO_ASSERT(bin_file, "Can't open bin file: \"" + m_binPath + "\"");
+        
+            // create xml file 
+            std::ofstream xml_file(m_xmlPath, std::ios::out);
+            xml_file.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+            OPENVINO_ASSERT(xml_file, "Can't open xml file: \"" + m_xmlPath + "\"");
+
             serializeFunc(xml_file, bin_file, model, m_version);
-        } catch (const ov::AssertFailure&) {
-            // optimization decision was made to create .bin file upfront and
-            // write to it directly instead of buffering its content in memory,
-            // hence we need to delete it here in case of failure
-            xml_file.close();
-            bin_file.close();
+        } catch (const ov::AssertFailure& e) {
```

Comment:
Could you cover case with `std::ofstream::failure` exception with some test?

Reply:
Given that I removed ```catch (std::ofstream::failure const& e)``` are you suggesting to write a test case for filled disk. If so I'm quite new to the repository and doesn't know how the file is exactly written, For a disk full condition, this can be tricky because actually filling a disk may not be practical for a test environment.

Are you suggesting we do something like this:
```
TEST_CASE("Test writing to full disk", "[file_write]") {
    // Its a sudo code
    try {
        std::ofstream bin_file("test.bin", std::ios::out | std::ios::binary);
        bin_file.exceptions(std::ofstream::failbit | std::ofstream::badbit);
        // Assuming "writeHugeAmountOfData" is a function that will exceed disk space. We can do something like 
        // creating a mock bin_file and throwing an error when say a few 100 bits are written
        writeHugeAmountOfData(bin_file); 
        // The line above should throw an exception; if it doesn't, the test should fail
        FAIL("Expected std::ofstream::failure due to disk space");
    } catch (const std::ofstream::failure& e) {
        // Assert: Verify that an exception due to disk space is caught
        SUCCEED("Caught std::ofstream::failure as expected");
    } catch (...) {
        // Any other exception means the test failed
        FAIL("Caught an unexpected exception type");
    }
}
```

Reply:
After doing this for bin and XML where do you want me to put the test?
```
std::ofstream bin_file;
bin_file.exceptions(std::ofstream::failbit | std::ofstream::badbit);
bin_file.open(m_binPath, std::ios::out | std::ios::binary);
OPENVINO_ASSERT(bin_file.is_open(), "Can't open bin file: \"" + m_binPath + "\"");
```

Reply:
@mishra-18 `createReadOnlyFile()` looks fine, about the test: `openvino/src/core/tests/pass/serialization/serialize.cpp` is the right place, but no need to change `SetUp()`, just use `TEST` macro instead of `TEST_P`, and path to .xml and .bin may be any (e.g. `CWD/file.xml`)

Reply:
Thanks for the review @vurusovs.
Wanted to make sure if ```std::shared_ptr<ov::Model> m = std::make_shared<ov::Model>();``` has been used correctly?
```
TEST(SerializationTest, WriteInReadOnly) {
    // Set up paths
    std::string test_xml_path = "test_file.xml";
    std::string test_bin_path = "test_file.bin";

    // Create read-only files
    ASSERT_TRUE(createReadOnlyFile(test_xml_path));
    ASSERT_TRUE(createReadOnlyFile(test_bin_path));

    // Set up the serializer with the current paths
    ov::pass::Serialize serializer(test_xml_path, test_bin_path);

    // Create a model or use a mock if you have one
    std::shared_ptr<ov::Model> m = std::make_shared<ov::Model>();

    // Expect that running the serializer on a read-only file throws an exception
    EXPECT_THROW(serializer.run_on_model(m), ov::AssertFailure);

    // Confirm that files were not successfully written.
    ASSERT_FALSE(std::ifstream(test_xml_path).good());
    ASSERT_FALSE(std::ifstream(test_bin_path).good());

    // Clean up
    std::remove(test_xml_path.c_str());
    std::remove(test_bin_path.c_str());
}
```

Reply:
Hi @mishra-18
ov::Model has no default constructor. You can create empty model with
```auto m = std::make_shared<ov::Model>(ov::OutputVector{}, ov::ParameterVector {}, "");```

Don't use please hardcoded file names. It may cause conflicts when tests are launched in parallel.
Use smth like that:
``` 
std::string filePrefix = ov::test::utils::generateTestFilePrefix();
m_out_xml_path = filePrefix + "test_file.xml";
m_out_bin_path = filePrefix + "test_file.bin";
```


Please add your test to openvino/src/core/tests/pass/serialization/serialize.cpp. It will be launched in CI